membrane_curvature/tests/test_mdakit_membcurv.py

```diff
@@ -144,44 +151,152 @@ def test_mean_curvature_all():
+    np.array([[0., 0., 150.],
+              [100., 0., 150.],
+              [200., 0., 150.],
+              [0., 100., 150.],
+              [100., 100., 120.],
+              [200., 100., 120.],
+              [0., 200., 120.],
+              [100., 200., 120.],
+              [200., 200., 120.]]),
+    # z_avg in grid
+    np.array(([150., 150., 120.],
+              [150., 120., 120.],
+              [150., 120., 120.]))
+)])
+def test_derive_surface_from_numpy(dummy_array, n_cells, max_width, z_avg):
+    u = mda.Universe(dummy_array, n_atoms=len(dummy_array))
+    selection = u.select_atoms('index 0:9')
+    max_width_x = max_width_y = max_width
+    surf = derive_surface(n_cells, selection, max_width_x, max_width_y)
+    for cell, calc_cell in zip(z_avg, surf):
+        assert_almost_equal(cell, calc_cell)
+
+
+@pytest.mark.parametrize('n_cells, max_width', [(3, 300.)])
+@pytest.mark.parametrize('dummy_array, z_avg', [(
+    # dummy coordinates [x,y,z]
+    np.array([[0., 0., 150.],
+              [0., 0., 150.],
+              [200., 0., 150.],
+              [0., 0., 150.],
+              [100., 100., 150.],
+              [200., 100., 150.],
+              [0., 200., 150.],
+              [100., 200., 150.],
+              [200., 200., 150.]]),
+    # z_avg in grid
+    np.array(([150., np.nan, 150.],
+              [np.nan, 150., 150.],
+              [150., 150., 150.]))
+)])
+def test_derive_surface_from_numpy_with_nans(dummy_array, n_cells, max_width, z_avg):
+    u = mda.Universe(dummy_array, n_atoms=len(dummy_array))
+    selection = u.select_atoms('index 0:9')
+    max_width_x = max_width_y = max_width
+    surf = derive_surface(n_cells, selection, max_width_x, max_width_y)
+    for cell, calc_cell in zip(z_avg, surf):
+        assert_almost_equal(cell, calc_cell)
+
+
+@pytest.mark.parametrize('x_bin, y_bin, x_range, y_range, z_surface', [
+
+    (3, 3, (0, 300), (0, 300), np.array(([150., np.nan, 150.],
+                                         [np.nan, 150., 150.],
+                                         [150., 150., 150.]))),
+    (3, 4, (0, 300), (0, 400), np.array([[150.,  np.nan, 150.,  np.nan],
+                                         [np.nan, 150., 150.,  np.nan],
+                                         [150., 150., 150.,  np.nan]]))
 ])
-def test_grid_map_grids(dummy_coord, factor):
-    cell = (dummy_coord[0] * factor, dummy_coord[1] * factor)
-    assert grid_map(dummy_coord, factor) == cell
-
-
-@pytest.mark.parametrize('dummy_coords, expected', [
-    # dummy coordinates (x,y)
-    (0, 0), (-1, 0), (2, 0),
-    (0, 1), (-1, 1), (2, 1),
-    (0, 2), (-1, 2), (2, 2),
-    # should map to
-    (0, 0), (2, 0), (2, 0),
-    (0, 1), (2, 1), (2, 1),
-    (0, 2), (2, 2), (2, 2)])
-@pytest.mark.xfail(reason='Incorrect mapping of negative coordinates')
-def test_grid_map_9grid_negative(dummy_coords, expected):
-    assert grid_map(dummy_coords, 1) == expected
+@pytest.mark.parametrize('dummy_array', [(
```

Comment:
This doesn't need to be parametrized either, as there's only one entry. Could you put `dummy_array` inside the test function?